### DISCUSSION TITLE
Merging two R6 microtables?

### DISCUSSION THREAD

**Discussion by HCB-M2RG** [ChiLiubio/microeco#298] [Q&A]
Hello, I have two data sets that together make up a whole set of samples. Both are pulled easily into Microeco from qiime2 outputs. Is there a simple way to merge the two R6 tables into a single microtable?

**Comment by ChiLiubio** [comment]
Good question. Currently, there is no a specific funtion for merging two microtable objects because the steps are not fixed for differernt steps. To do this, you can directly add each table into the corresponding table in another object.

**Reply by HCB-M2RG** [reply]
Hello, thanks...
I tried this but am conceptually challenged. I attempeted to do an rbind on the sample_table, tax_table and otu_table. I thought this would work given that they are dataframes but the otu_tables do not have matching numbers of columns. Also, I am not sure how to add the phylo_tree objects together because they are not data frames.

Here is my code... I am working with two sets of samples comming from two sequencing plates. Samples from the study are shared across both plates. I can upload dataset_16S_A and dataset_16S_B individually and work with them within the package.

dataset_16S_tot <- microtable$new(
  sample_table = rbind(dataset_16S_A$sample_table, dataset_16S_B$sample_table),
  tax_table = rbind(dataset_16S_A$tax_table, dataset_16S_B$tax_table),
  otu_table = rbind(dataset_16S_A$otu_table, dataset_16S_B$otu_table),
  phylo_tree = rbind(dataset_16S_A$phylo_tree, dataset_16S_B$phylo_tree)
)
	dataset_16S_tot

**Comment by ChiLiubio** [comment]
Hi. Please first generate each table successfully, then generate microtable object. For example, `new_sample_table = rbind(dataset_16S_A$sample_table, dataset_16S_B$sample_table)` and `View(new_sample_table)` to check whether it is correct. I am confused that If you use rbind to combine samples of A and B, it means that samples from A and B are different samples. But in otu_table, they will be same samples as rbind only merge OTU/ASVs. Phylo tree can not be merged directly.
